crates/rolldown/src/utils/chunk/determine_export_mode.rs

```diff
@@ -0,0 +1,44 @@
+use rolldown_common::{EcmaModule, OutputExports, SymbolRef};
+use rolldown_error::{BuildDiagnostic, DiagnosableResult};
+use rolldown_rstr::Rstr;
+
+// Port from https://github.com/rollup/rollup/blob/master/src/utils/getExportMode.ts
+pub fn determine_export_mode(
+  export_mode: &OutputExports,
+  module: &EcmaModule,
+  exports: &[(Rstr, SymbolRef)],
+) -> DiagnosableResult<OutputExports> {
+  match export_mode {
+    OutputExports::Named => Ok(OutputExports::Named),
+    OutputExports::Default => {
+      if exports.len() != 1 || exports[0].0.as_str() != "default" {
+        return Err(vec![BuildDiagnostic::invalid_export_option(
```

Comment:
New `BuildDiagnostic` should have a test in `tests/rolldown/errors or warnings`.

Reply:
I moved the `determine_export_mode` from the `render_chunk_exports` to a splitter folder, as it won't use this function in the `render_chunk_exports`.